lib/improver/cli/nbhood.py

```diff
@@ -347,52 +141,37 @@ def process(cube, neighbourhood_output, neighbourhood_shape, radius=None,
             neighbourhood_output='percentiles'.
         ValueError:
             If degree_as_complex is used with neighbourhood_shape='circular'.
-        ValueError:
-            If degree_as_complex is used with apply_recursive_filter.
-        ValueError:
-            If neighbourhood_shape is 'circular' and apply_recursive_function
-            is True.
-
     """
-    if (neighbourhood_output == "percentiles" and
-            neighbourhood_shape == "square"):
-        raise RuntimeError('neighbourhood_shape="square" cannot be used with'
-                           'neighbourhood_output="percentiles"')
-
-    if neighbourhood_output == "percentiles" and weighted_mode:
-        raise RuntimeError('weighted_mode cannot be used with'
-                           'neighbourhood_output="percentiles"')
-
-    if (neighbourhood_output == "probabilities" and
-            percentiles != DEFAULT_PERCENTILES):
-        raise RuntimeError('percentiles cannot be DEFAULT_PERCENTILES with'
-                           'neighbourhood_output="probabilities"')
-
-    if mask_cube and neighbourhood_shape == "circular":
-        raise RuntimeError('mask_cube cannot be used with'
-                           'neighbourhood_output="circular"')
-
-    if neighbourhood_shape == 'circular' and apply_recursive_filter:
-        raise ValueError('Recursive filter option is not applicable to '
-                         'circular neighbourhoods. ')
-
-    if degrees_as_complex:
-        if neighbourhood_output == "percentiles":
-            raise ValueError(
-                'Cannot generate percentiles from complex numbers')
-        if neighbourhood_shape == "circular":
+    from improver.nbhood import radius_by_lead_time
+    from improver.nbhood.nbhood import (
+        GeneratePercentilesFromANeighbourhood, NeighbourhoodProcessing)
+    from improver.utilities.pad_spatial import remove_cube_halo
+    from improver.wind_calculations.wind_direction import WindDirection
+
+    if neighbourhood_output == "percentiles":
+        if neighbourhood_shape == "square":
+            raise RuntimeError('neighbourhood_shape="square" cannot be used '
+                               'with neighbourhood_output="percentiles"')
```

Comment:
A similar error will be raised from the plugin - we don't need this here.

Reply:
Good spot.